crates/database/src/fee_policies.rs

```diff
@@ -52,24 +53,40 @@ pub async fn insert_batch(
     query_builder.build().execute(ex).await.map(|_| ())
 }
 
-pub async fn fetch(
+pub async fn fetch_all(
     ex: &mut PgConnection,
-    auction_id: AuctionId,
-    order_uid: OrderUid,
-) -> Result<Vec<FeePolicy>, sqlx::Error> {
-    const QUERY: &str = r#"
-    SELECT * FROM fee_policies
-    WHERE auction_id = $1 AND order_uid = $2
-    ORDER BY application_order
-"#;
-    let rows = sqlx::query_as::<_, FeePolicy>(QUERY)
-        .bind(auction_id)
-        .bind(order_uid)
-        .fetch_all(ex)
-        .await?
-        .into_iter()
-        .collect();
-    Ok(rows)
+    keys_filter: &[(AuctionId, OrderUid)],
+) -> Result<HashMap<(AuctionId, OrderUid), Vec<FeePolicy>>, sqlx::Error> {
+    if keys_filter.is_empty() {
+        return Ok(HashMap::new());
+    }
+
+    let mut query_builder = QueryBuilder::new("SELECT * FROM fee_policies WHERE ");
+    for (i, (auction_id, order_uid)) in keys_filter.iter().enumerate() {
+        if i > 0 {
+            query_builder.push(" OR ");
+        }
+        query_builder
+            .push("(")
+            .push("auction_id = ")
+            .push_bind(auction_id)
+            .push(" AND ")
+            .push("order_uid = ")
+            .push_bind(order_uid)
+            .push(")");
+    }
+
+    query_builder.push(" ORDER BY auction_id, order_uid, application_order");
```

Comment:
We only need to sort by `application_order` which could speed up the query depending on the available indices.